services/rpcfilters/hash_filter.go

```diff
@@ -0,0 +1,46 @@
+package rpcfilters
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type hashFilter struct {
+	hashes []common.Hash
+	mu     sync.Mutex
+	done   chan struct{}
+	timer  *time.Timer
```

Comment:
When is `timer` defined? It is not accepted in the constructor.

Reply:
i mentioned in the PR that timer is not used for hash filter only for log filter. it kinda preserves previous behavior. i can create a followup PR to pass it in the constructor